ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java

```diff
@@ -114,6 +115,9 @@ private Router setupRouter() {
     final Router router = Router.router(vertx);
 
     router.route().handler(new LoggingHandler(server, loggingRateLimiter));
+    if (server.getConfig().getBoolean(KsqlRestConfig.KSQL_SERVER_SNI_CHECK_ENABLE)) {
```

Comment:
In case we need to disable the check.

Reply:
It's too bad that this flag is a server-only config. It'd be nice if users had a way to override this (at the individual request level) but adding that mechanism for this change is probably overkill.